code/modules/admin/admin.dm

```diff
@@ -842,3 +842,83 @@
+	var/department = input("Choose a fax", "Fax") as null|anything in GLOB.alldepartments
+	for(var/obj/machinery/photocopier/faxmachine/sendto in GLOB.allfaxes)
+		if(sendto.department == department)
+
+			if (!istype(src,/datum/admins))
+				src = usr.client.holder
+			if (!istype(src,/datum/admins))
+				to_chat(usr, "Error: you are not an admin!")
+				return
+
+			var/replyorigin = input(src.owner, "Please specify who the fax is coming from", "Origin") as text|null
+
+			var/obj/item/weapon/paper/admin/P = new /obj/item/weapon/paper/admin( null ) //hopefully the null loc won't cause trouble for us
+			faxreply = P
+
+			P.admindatum = src
+			P.origin = replyorigin
+			P.destination = sendto
+
+			P.adminbrowse()
+
+/datum/admins/var/obj/item/weapon/paper/admin/faxreply // var to hold fax replies in
+
+/datum/admins/proc/faxCallback(var/obj/item/weapon/paper/admin/P, var/obj/machinery/photocopier/faxmachine/destination)
+	var/customname = input(src.owner, "Pick a title for the report", "Title") as text|null
+
+	P.name = "[P.origin] - [customname]"
+	P.desc = "This is a paper titled '" + P.name + "'."
+
+	var/shouldStamp = 1
+	if(!P.sender) // admin initiated
+		switch(alert("Would you like the fax stamped?",, "Yes", "No"))
+			if("No")
+				shouldStamp = 0
+
+	if(shouldStamp)
+		P.stamps += "<hr><i>This paper has been stamped by the [P.origin] Quantum Relay.</i>"
+
+		var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+		var/{x; y;}
+		x = rand(-2, 0)
+		y = rand(-1, 2)
+		stampoverlay.pixel_x = x
+		stampoverlay.pixel_y = y
+		stampoverlay.icon_state = "paper_stamp-cent"
+
+		if(!P.stamped)
+			P.stamped = new
+		P.stamped += /obj/item/weapon/stamp/centcomm
+		P.overlays += stampoverlay
+
+	var/obj/item/rcvdcopy
+	rcvdcopy = destination.copy(P)
+	rcvdcopy.loc = null //hopefully this shouldn't cause trouble
+	GLOB.adminfaxes += rcvdcopy
+
+	if(destination.recievefax(P))
+		to_chat(src.owner, "<span class='notice'>Message reply to transmitted successfully.</span>")
+		if(P.sender) // sent as a reply
+			log_admin("[key_name(src.owner)] replied to a fax message from [key_name(P.sender)]")
+			for(var/client/C in GLOB.admins)
+				if(check_rights(R_ADMIN))
+					to_chat(C, "<span class='log_message'><span class='prefix'>FAX LOG:</span>[key_name_admin(src.owner)] replied to a fax message from [key_name_admin(P.sender)] (<a href='?_src_=holder;AdminFaxView=\ref[rcvdcopy]'>VIEW</a>)</span>")
+		else
+			log_admin("[key_name(src.owner)] has sent a fax message to [destination.department]")
+			for(var/client/C in GLOB.admins)
+				if(check_rights(R_ADMIN))
+					to_chat(C, "<span class='log_message'><span class='prefix'>FAX LOG:</span>[key_name_admin(src.owner)] has sent a fax message to [destination.department] (<a href='?_src_=holder;AdminFaxView=\ref[rcvdcopy]'>VIEW</a>)</span>")
+
+	else
+		to_chat(src.owner, "<span class='warning'>Message reply failed.</span>")
+
+	spawn(100)
+		qdel(P)
+		faxreply = null

```

Comment:
If P is a temporary object then it should be deleted in the same namespace that it's created in. Deleting P in the proc it's passed to is unintuitive.

Spawn is also a problem here but a relatively minor one.